test/cloudtest/test_storaged_shared_fate.py

```diff
@@ -0,0 +1,219 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import subprocess
+import time
+from textwrap import dedent
+
+from materialize.cloudtest.application import MaterializeApplication
+
+CLUSTER_SIZE = 8
+NUM_SOURCES = 4
+
+
+def populate(mz: MaterializeApplication, seed: int) -> None:
+    create_sources = "\n".join(
+        f"""
+            > CREATE SOURCE s{i}
+              IN CLUSTER storaged_shared_fate
+              FROM KAFKA CONNECTION kafka (TOPIC 'testdrive-storaged-shared-fate-${{testdrive.seed}}')
+              FORMAT BYTES
+              ENVELOPE NONE;
+    """
+        for i in range(NUM_SOURCES)
+    )
+
+    create_views = "\n".join(
+        f"""
+            > CREATE MATERIALIZED VIEW v{i} AS SELECT COUNT(*) FROM t1 UNION ALL SELECT COUNT(*) FROM s{i};
+    """
+        for i in range(NUM_SOURCES)
+    )
+
+    create_indexes_and_counts = "\n".join(
+        f"""
+            > CREATE DEFAULT INDEX ON v{i};
```

Comment:
There is no need to create an index here. An index is is a compute-side concept but here you are using the materialized view to simply validate your sources, you are not disrupting the compute side in this test.